src/components/Optimize.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { Component, ChangeEvent } from "react"
+import ReactTooltip from "react-tooltip"
+
+import CLASSES from "../constants/classes"
+import { ISettingsElement } from "../constants/interfaces"
+
+interface MyProps {
+    optimizeSettings: Array<ISettingsElement>
+    updateOptimize: (fieldKey: string, fieldValue: number) => void
+    applyOpitimization: (optimizationList: string[]) => void
+}
+
+interface MyState {
+    show: boolean
+    tooltipText: string | JSX.Element
+}
+
+export default class Optimize extends Component<MyProps, MyState> {
+    /**
+     * A small settings menu to enable and disable things
+     * And to be able to adjust certain sizes
+     */
+    constructor(props: MyProps) {
+        super(props)
+        this.state = {
+            show: false,
+            tooltipText: "",
+        }
+    }
+
+    showSettings = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        this.setState({
+            show: true,
+        })
+    }
+
+    hideSettings = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        this.setState({
+            show: false,
+        })
+    }
+
+    onChange = (e: ChangeEvent<HTMLInputElement>, itemShortName: string) => {
+        const newValue = parseFloat(e.target.value)
+        this.props.updateOptimize(itemShortName, newValue)
+    }
+
+    onMouseEnter = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, item: JSX.Element) => {
+        this.setState({
+            tooltipText: item,
+        })
+    }
+
+    onApply = (
+        e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+        optimizationName: string | number
+    ) => {
+        this.props.applyOpitimization(["" + optimizationName])
```

Comment:
Here you are just converting number to string, I think
Perhaps format string would make it easier to read:
```this.props.applyOpitimization([`${optimizationName}`])```

Reply:
I changed it, thanks for the advice!